packages/suite/src/actions/wallet/createPassphraseFlowManager.ts

```diff
@@ -0,0 +1,876 @@
+        }
+    };
+
+    const _invalidateAndForgetAddedDevice = (device: TrezorDevice) => {
+        const settings = selectSuiteSettings(getState());
+
+        dispatch(deviceActions.forgetDevice({ device, settings }));
+
+        const newDevice = selectSelectedDevice(getState());
+        dispatch(deviceActions.selectDevice(newDevice));
+        discoveryHook.deregisterAuthHook(device);
+    };
+
+    const _requestPassphraseWalletCreation = async (
+        device: TrezorDevice,
+        deviceParams: Parameters<typeof TrezorConnect.getDeviceState>[0],
+    ): Promise<
+        | {
+              result: 'success';
+          }
+        | {
+              result: 'error';
+              error:
+                  | 'auth-failed'
+                  | 'passphrase-duplicate'
+                  | 'device-not-ready'
+                  | 'enter-passphrase-cancel'
+                  | 'enter-passphrase-back';
+          }
+    > => {
+        const { deviceReadyPromise, responsePromise } = await trezorConnectService.invokeUIAction(
+            trezorConnect => {
+                const result = trezorConnect.getDeviceState(deviceParams);
+
+                return result;
+            },
+        );
+
+        deviceReadyPromise.then(() => {
+            dispatch(setPassphraseFlowTransactionLoading({ loading: false }));
+        });
+
+        const response = await responsePromise;
+
+        if (response.success) {
+            const { state, _state } = response.payload;
+            const s = state.split(':')[0];
+            const devices = selectDevices(getState());
+            const duplicate = devices?.find(
+                d =>
+                    d.state?.staticSessionId &&
+                    d.state.staticSessionId.split(':')[0] === s &&
+                    d.instance !== device.instance,
+            );
+
+            if (duplicate) {
+                // Use the notificationsActions to show a notification instead of a modal
+                dispatch(
+                    notificationsActions.addToast({
+                        type: 'auth-failed',
+                        error: `Passphrase already exists for device ${device.label || 'Unknown'}`,
+                    }),
+                );
+
+                dispatch(
+                    authorizeDeviceThunkRejected({
+                        error: 'passphrase-duplicate',
+                        device,
+                    }),
+                );
+
+                return {
+                    result: 'error',
+                    error: 'passphrase-duplicate',
+                };
+            }
+
+            const freshDevice = getSelectedDevice(device, selectDevices(getState()));
+
+            if (!freshDevice) {
+                throw new Error('device-not-selected');
+            }
+
+            // NOTE: triggers the discovery
+            dispatch(
+                authorizeDeviceThunkFulfilled({
```

Comment:
@martykan says:
Remove this in favor of direct dispatch of the discovery. 

Then, remove the helper that creates this type.